### PR TITLE
Feature/poky update

### PR DESCRIPTION
**Pull request by skovbakke**
Update poky to the new release honister. The old release hardknott is EOL in March. Updating includes most of our upstreamed patches, systemd v249, that we are already using, as well as kubernetes: v1.22

The PR goes along with https://github.com/distributed-technologies/mukube-configurator/pull/42

### CODE REVIEW

File: docs/patches.md [hunk 1]
```diff
@@ -4,34 +4,24 @@ All changes upstreamed, submitted upstream or cherry-picked from upstream should
...
+| [`0001-CRI-O-Remove-containerd-as-RDEPENDS.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Remove-containerd-as-RDEPENDS.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Removes containerd as a dependency in cri-o |
+| [`0001-CRI-O-Update-to-1.22.1-and-fix-SRCREV-SRC_URI.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Update-to-1.22.1-and-fix-SRCREV-SRC_URI.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Fixes a misconfiguration in the CRI-O SRCREV/SRC_URI |
```
Comment (klausenbusk) [inline docs/patches.md:12]: This sounds like something upstream should fix.

It is already [fixed in master](https://git.yoctoproject.org/meta-virtualization/commit/recipes-containers/cri-o/cri-o_git.bb?id=17e0be0dc3763bd7a92728b90087072c5afc59ea) so just a missing backport it seems.

Let me try to reach out to upstream.

Comment (klausenbusk) [reply docs/patches.md:12]: Backport submitted: https://lists.yoctoproject.org/g/meta-virtualization/topic/honister_patch_cri_o/87765433

Comment (klausenbusk) [reply docs/patches.md:12]: Cherry-picked: https://git.yoctoproject.org/meta-virtualization/commit/?h=honister&id=7618d5046a24e4c8852560b17f815b1ce3cba160

File: docs/patches.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ All changes upstreamed, submitted upstream or cherry-picked from upstream should
...
 | [`0001-CRI-O-Remove-containerd-as-RDEPENDS.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Remove-containerd-as-RDEPENDS.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Removes containerd as a dependency in cri-o |
 | [`0001-CRI-O-Update-to-1.22.1-and-fix-SRCREV-SRC_URI.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Update-to-1.22.1-and-fix-SRCREV-SRC_URI.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Fixes a misconfiguration in the CRI-O SRCREV/SRC_URI |
+| [`0001-dm-verity-img.bbclass-Fix-the-override-fixups.patch`](../patches/meta-security/misc/0001-dm-verity-img.bbclass-Fix-the-override-fixups.patch) | [meta-security](https://git.yoctoproject.org/cgit/cgit.cgi/meta-security/) | local patch | Fixes the override syntax |
```
Comment (klausenbusk) [inline docs/patches.md:13]: Submitted upstream: https://lists.yoctoproject.org/g/yocto/topic/meta_security_patch/87786381. I expect it to be backported.

Comment (klausenbusk) [reply docs/patches.md:13]: Landed upstream: https://git.yoctoproject.org/meta-security/commit/?id=126860dac30d50a5894588b874ae917b0b310142, please cherry-pick.

File: docs/patches.md (hunk 1 above)
Comment (klausenbusk) [reply docs/patches.md:12]: Please remove the patch as it has been backported.

File: docs/patches.md [hunk 3]
```diff
@@ -4,34 +4,24 @@ All changes upstreamed, submitted upstream or cherry-picked from upstream should
...
+| [`0001-CRI-O-Remove-containerd-as-RDEPENDS.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Remove-containerd-as-RDEPENDS.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Removes containerd as a dependency in cri-o |
+| [0001-dm-verity-img.bbclass-Fix-wrong-override-syntax-for-.patch](../patches/meta-security/misc/0001-dm-verity-img.bbclass-Fix-wrong-override-syntax-for-.patch) | [meta-security](https://git.yoctoproject.org/cgit/cgit.cgi/meta-security/) | [upstreamed](https://git.yoctoproject.org/meta-security/commit/?id=126860dac30d50a5894588b874ae917b0b310142), cherry-picked | Fixes the override syntax |
```
Comment (klausenbusk) [inline docs/patches.md:12]: I have requested a backport: https://lists.yoctoproject.org/g/yocto/topic/honister_patch/88643459

File: docs/patches.md [hunk 4]
```diff
@@ -4,34 +4,24 @@ All changes upstreamed, submitted upstream or cherry-picked from upstream should
...
 | [`0001-repart-Support-volatile-root-for-finding-the-root-pa.patch`](../meta-k8s-setup/recipes-core/systemd/systemd/0001-repart-Support-volatile-root-for-finding-the-root-pa.patch) | [systemd](https://github.com/systemd/systemd) | [upstreamed](https://github.com/systemd/systemd/commit/54632d2ea405e9df9ca9c3c469c7d0feb3a85323) | Make [`systemd-repart`](https://www.freedesktop.org/software/systemd/man/systemd-repart.html) work with `/run/systemd/volatile-root`. This is needed as systemd can't find the original root partition due to [`ramrootfs`](../meta-k8s-setup/recipes-core/initrdscripts/initramfs-framework/ramrootfs) |
+| [`0001-CRI-O-Remove-containerd-as-RDEPENDS.patch`](../patches/meta-virtualization/misc/0001-CRI-O-Remove-containerd-as-RDEPENDS.patch) | [meta-virtualization](https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/) | local patch | Removes containerd as a dependency in cri-o |
```
Comment (klausenbusk) [inline docs/patches.md:11]: Could this just be a `bbappend` file?

**Review by jacobcrawford — CHANGES_REQUESTED** [review]
Please paste a screenshot of a successful test run.

**Comment by jacobcrawford** [comment]
Do we need to take into account the KubeletConfiguration generated in mukube-configurator?

**Comment by jacobcrawford** [comment]
When building this on a fresh Debian I get:
 
image

If there are hosttools needed we should list these requirements.

**Comment by skovbakke** [comment]
> When building this on a fresh Debian I get:
> 
> image
> 
> If there are hosttools needed we should list these requirements.

Indeed. This is documented, [here](https://docs.yoctoproject.org/ref-manual/system-requirements.html#required-packages-for-the-build-host), as these are part of the requirements to using yocto.

**Comment by skovbakke** [comment]
> Do we need to take into account the KubeletConfiguration generated in mukube-configurator?

No, that change does not affect anything here.

**Comment by skovbakke** [comment]
All test are passing
image

File: docs/patches.md (hunk 4 above)
Comment (skovbakke) [reply docs/patches.md]: Yes, that is a good idea.

**Comment by jacobcrawford** [comment]
image

Some recipes are not able to be fetched. What do we do when this happens?

**Comment by jacobcrawford** [comment]
image

**Comment by jacobcrawford** [comment]
image

**Comment by skovbakke** [comment]
> image
> 
> Some recipes are not able to be fetched. What do we do when this happens?

We try to fix it. It turns out that the branch name has changed from `master` to `main`. So in order to fetch the source code we needed to specify that the branch is main (as master is the default).  The error did not show up for me, probably because the source was already fetched and/or the step/recipe was cached.

**Comment by jacobcrawford** [comment]
If the master branch has changed name we should make the main branch the one we use.
setting
`SRC_URI = "git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git;protocol=https;branch=main"`
in meta-k8s-setup/recipes-core/microcode/amd-microcode_20210919.bb worked for me.

**Review by jacobcrawford — CHANGES_REQUESTED** [review]
If we update the branch for microcode to `main`, I think this is good to go :)

**Review by klausenbusk — APPROVED** [review]
LGTM 🎉

Tested with https://github.com/distributed-technologies/mukube-configurator/pull/42 with https://github.com/distributed-technologies/mukube-configurator/pull/43 and https://github.com/distributed-technologies/mukube-configurator/pull/44 on top.

**Review by klausenbusk — DISMISSED** [review]
LGTM 🎉

Tested with https://github.com/distributed-technologies/mukube-configurator/pull/42 with https://github.com/distributed-technologies/mukube-configurator/pull/43 and https://github.com/distributed-technologies/mukube-configurator/pull/44 on top.

**Review by jacobcrawford — APPROVED** [review]
LGTM!

**Review by klausenbusk — APPROVED** [review]
LGTM 🎉

Tested with distributed-technologies/mukube-configurator#42 with distributed-technologies/mukube-configurator#43 and distributed-technologies/mukube-configurator#44 on top.

**Pull request merged**